From 3cdba40988549d38647e0650ceb18a18a05ca04f Mon Sep 17 00:00:00 2001
From: Boris Ranto <branto@redhat.com>
Date: Fri, 6 Nov 2015 13:00:45 +0100
Subject: [PATCH 08/11] ipmitool-1.8.11-no-work-setaccess.patch

RESOLVES: #903251 - link=on and ipmi=on no longer work for setaccess

Backported by: phracek
---
 lib/ipmi_channel.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/ipmi_channel.c b/lib/ipmi_channel.c
index 634b4d1..fa68e35 100644
--- a/lib/ipmi_channel.c
+++ b/lib/ipmi_channel.c
@@ -500,10 +500,10 @@ ipmi_set_user_access(struct ipmi_intf * intf, int argc, char ** argv)
 			set_access.callin_callback = !(strncmp (argv[i]+7, "off", 3));
 		}
 		else if (strncmp(argv[i], "link=", 5) == 0) {
-			set_access.link_auth = strncmp (argv[i]+5, "off", 3);
+			set_access.link_auth = (strncmp (argv[i]+5, "off", 3) != 0);
 		}
 		else if (strncmp(argv[i], "ipmi=", 5) == 0) {
-			set_access.ipmi_messaging = strncmp (argv[i]+5, "off", 3);
+			set_access.ipmi_messaging = (strncmp (argv[i]+5, "off", 3) != 0);
 		}
 		else if (strncmp(argv[i], "privilege=", 10) == 0) {
 			if (str2uchar(argv[i]+10, &privilege_limit) != 0) {
-- 
2.1.0

